### PR TITLE
Refactor the fsspec to message functions

### PR DESCRIPTION
**Pull request by mraspaud**
This PR refactors the fsspec to message functions to make them more easily reusable.

@paulovcmedeiros you might be interested in this.

This needs https://github.com/fsspec/filesystem_spec/pull/950 to work fully

### CODE REVIEW

**Review by pnuu — APPROVED** [review]
LGTM, feel free to merge when the fsspec PR is merged and released so this can be actually used.

**Comment by pnuu** [comment]
Please remind me, what was the use for the `filesystem` part of the message dictionary? As far as I can see it isn't needed by `fsspec`:
```python
uri = "tar://tmp/bar.txt::ssh://127.0.0.1/tmp/test.tar"
fsfiles = fsspec.open_files(uri)
print(fsfiles[0])
# <OpenFile 'tmp/bar.txt'>
```
And what does `fo` stand for? If "output file", then isn't the tar file actually the input where the data is coming from and then stored locally as `uid`?

**Comment by mraspaud** [comment]
the `filesystem` item is just here to make sure we have everything. Indeed, if we were to include eg `filecache`, the specific options for it can't be added to the uri, but need to be provided as storage options inside `filesystem`. Now, it is true that the implemented functions do not add filecache yet...

Regarding `fo`, from what I gather it means `file object`.

**Comment by mraspaud** [comment]
@pnuu there was a significant change in the last commit, maybe you want to have a look at it again.

File: pytroll_collectors/fsspec_to_message.py [hunk 1]
```diff
@@ -25,31 +26,50 @@ def create_message(fs, file, subject, metadata=None):
...
+    return uri
+
+
+def create_netloc(ssh):
+    """Create the netloc from ssh parameters like `host`, `port`, `username`, `password`."""
+    host = ssh.get("host")
+    if host is None:
+        return ""
+    username = ssh.get("username")
+    password = ssh.get("password")
```
Comment (pnuu) [inline pytroll_collectors/fsspec_to_message.py:59]: I hope this doesn't imply that the credentials are passed in plain text in the messages everyone within the network can see...?

File: pytroll_collectors/fsspec_to_message.py [hunk 2]
```diff
@@ -25,31 +26,50 @@ def create_message(fs, file, subject, metadata=None):
...
-    uri = protocol + '://' + file['name']
-    return uri
+    netloc = create_netloc(loaded_fs)
+    uid = protocol + '://' + netloc + file['name']
+    return uid
 
 
-def _create_uid_from_uri(uri, loaded_fs):
-    uid = uri
+def _create_uid_from_uri(uid, loaded_fs):
```
Comment (pnuu) [inline pytroll_collectors/fsspec_to_message.py:45]: This is creating URI from UID, so the naming is reversed

File: pytroll_collectors/fsspec_to_message.py [hunk 3]
```diff
@@ -84,7 +104,34 @@ def extract_files_to_message(file, fs, subject, packing=None):
...
+    if packing is None:
+        cls = get_filesystem_class(target_protocol or "ssh")
+        cls = ".".join((cls.__module__, cls.__name__))
+        target_options = target_options or dict(host=socket.gethostname())
+        fs_dict = dict(cls=cls,
+                       protocol=target_protocol or "ssh",
+                       args=[],
+                       **target_options
+                       )
+        file = dict(name=filename)
```
Comment (pnuu) [inline pytroll_collectors/fsspec_to_message.py:125]: This is not tested

**Review by pnuu — CHANGES_REQUESTED** [review]
One misnamed function, one security concern and some lines the tests don't cover.

File: pytroll_collectors/fsspec_to_message.py (hunk 3 above)
Comment (mraspaud) [reply pytroll_collectors/fsspec_to_message.py]: Fixed

File: pytroll_collectors/fsspec_to_message.py (hunk 2 above)
Comment (mraspaud) [reply pytroll_collectors/fsspec_to_message.py]: Fixed

**Review by pnuu — APPROVED** [review]
LGTM

**Pull request merged**